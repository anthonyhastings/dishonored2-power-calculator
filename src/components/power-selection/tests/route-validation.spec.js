import React from 'react';
import Immutable from 'immutable';
import renderer from 'react-test-renderer';
import PowerSelectionRouteValidation from '../route-validation';

jest.mock('Components/page-not-found/lazy', () => 'MockPageNotFound');

jest.mock('../container', () => 'MockPowerSelection');

describe('PowerSelectionRouteValidation component', () => {
  let testContext;

  const renderComponent = (props = {}) => (
    <PowerSelectionRouteValidation {...testContext.defaultProps} {...props} />
  );

  beforeEach(() => {
    testContext = {};

    testContext.defaultProps = {};
  });

  describe('when character is missing', () => {
    beforeEach(() => {
      testContext.component = renderer.create(renderComponent());
    });

    it('renders the PageNotFound component', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when character is present', () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        renderComponent({
          character: Immutable.Map(),
        })
      );
    });

    it('renders the PowerSelection component', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });
});
