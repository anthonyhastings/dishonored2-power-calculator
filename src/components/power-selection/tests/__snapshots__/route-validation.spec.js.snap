// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerSelectionRouteValidation component when character is missing renders the PageNotFound component 1`] = `<MockPageNotFound />`;

exports[`PowerSelectionRouteValidation component when character is present renders the PowerSelection component 1`] = `<MockPowerSelection />`;
